src/Model_Hydro/CPU_Hydro/CPU_Shared_FluUtility.cpp

```diff
@@ -596,12 +599,12 @@ bool Hydro_CheckUnphysical( const CheckUnphysical_t Mode, const real Fields[], c
 //       check discriminant for SRHD
 //       --> positive if and only if Eq. 15 in "Tseng et al. 2021, MNRAS, 504, 3298" has a positive root
 #        ifdef SRHD
-         const real Msqr         = SQR(Fields[MOMX]) + SQR(Fields[MOMY]) + SQR(Fields[MOMZ]);
-         const real Dsqr         = SQR(Fields[DENS]);
-         const real E_D          = Fields[ENGY] / Fields[DENS];
-         const real M_D          = SQRT( Msqr / Dsqr );
-         const real Temp         = SQRT( E_D*E_D + (real)2.0*E_D );
-         const real Discriminant = ( Temp + M_D )*( Temp - M_D );
+         real Msqr         = SQR(Fields[MOMX]) + SQR(Fields[MOMY]) + SQR(Fields[MOMZ]);
+         real Dsqr         = SQR(Fields[DENS]);
+         real E_D          = Fields[ENGY] / Fields[DENS];
+         real M_D          = SQRT( Msqr / Dsqr );
+         real Temp         = SQRT( E_D*E_D + (real)2.0*E_D );
+         real Discriminant = ( Temp + M_D )*( Temp - M_D );
```

Comment:
There are double declaration of `real` in lines 602-607 and 544. Other than that, I have no comment. Thanks for the refactoring!

Reply:
@zengbs Good catch. Thanks!